### PR TITLE
Sh view categories

### PR DESCRIPTION
**Pull request by SaraHackle**
#### Changes Made
1. Added file `Category.cs` class to `models` directory.
2. Added `CategoryRepository.cs` and `ICategoryRepository.cs` to `repositories` directory.
3. Added `categoryManager.js` in `modules` folder within `client` directory.
4. Added `Category.js` , and `CategoryList.js` in `components` folder within `client` directory
5. Modified `Header.js` to include a link to Categories page
6. Modified `ApplicationViews.js` to include a route to Categories page. 
7. Merged new main, fixed merge conflicts.

​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout SH-ViewCategories
    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
8. Test app functionality.
    > When a user navigates to the Categories link, they should see a list of all categories
9. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

File: Tabloid/client/src/components/CategoryList.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const getCategories = () => {
+    getAllCategories().then((categories) => setCategories(categories));
+  };
+
+
+  useEffect(() => {
+    getCategories();
+  }, []);
+
+  console.log(categories)
```
Comment (cjgochanour) [inline Tabloid/client/src/components/CategoryList.js]: I can see this console.log being useful for testing purposes but unnecessary for deployment.

**Review by cjgochanour — APPROVED** [review]
Works great! Thanks for removing the console.log

File: Tabloid/Controllers/CategoryController.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Tabloid.Models;
+using Tabloid.Repositories;
+
+namespace Tabloid.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
```
Comment (joshdbarton) [inline Tabloid/Controllers/CategoryController.cs:15]: Should this controller be auth protected? do we need any other auth checks in any of the routes?

File: Tabloid/Repositories/CategoryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                    cmd.CommandText = "SELECT id, name FROM Category Order By Name";
+                    var reader = cmd.ExecuteReader();
+
+                    var categories = new List<Category>();
+
+                    while (reader.Read())
+                    {
+                        categories.Add(new Category()
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
```
Comment (joshdbarton) [inline Tabloid/Repositories/CategoryRepository.cs]: Are there DbUtils methods that could be used in this repo to do some of this work? We'll want to use those utils methods consistently across the codebase.

File: Tabloid/Repositories/CategoryRepository.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            {
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"
+                        SELECT *
+                        FROM Category                    
+                        WHERE Id = @id                   
+                    ";
+                    cmd.Parameters.AddWithValue("@id", id);
```
Comment (joshdbarton) [inline Tabloid/Repositories/CategoryRepository.cs]: here's another example where the utils can help us.

File: Tabloid/client/src/components/ApplicationViews.js [hunk 5]
```diff
@@ -24,7 +25,12 @@ export default function ApplicationViews({ isLoggedIn }) {
                 <Route path="/register">
                     <Register />
                 </Route>
-            </Switch>
-        </main>
-    );
-}
+                
+                <Route path="/categories">
```
Comment (joshdbarton) [inline Tabloid/client/src/components/ApplicationViews.js]: should this be an `exact` match?

File: Tabloid/client/src/modules/categoryManager.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const baseUrl = "/api/Category";
+
+export const getAllCategories = () => {
```
Comment (joshdbarton) [inline Tabloid/client/src/modules/categoryManager.js:6]: if this route is auth protected, you'll need to get an api token to pass in to this fetch call.

**Review by joshdbarton — CHANGES_REQUESTED** [review]
take a look at my comments

**Review by joshdbarton — APPROVED** [review]

**Pull request merged**